src/zcash/cache.h

```diff
@@ -31,7 +31,7 @@ class BundleCacheHasher
     {
         static_assert(hash_select < 8, "BundleCacheHasher only has 8 hashes available.");
         uint32_t u;
-        std::memcpy(&u, key.begin() + 4 * hash_select, 4);
+        std::memcpy(&u, &*(key.begin() + 4 * hash_select), 4);
```

Comment:
I don't know why I used `std::array::begin` here when I wrote this; it should have been [`std::array::data`](https://en.cppreference.com/w/cpp/container/array/data). Can you test this, and if it works then change the commit to do this:
```suggestion
        std::memcpy(&u, key.data() + 4 * hash_select, 4);
```

Reply:
Yep, that works!